BTCPayServer/Plugins/PluginBuilderClient.cs

```
@@ -85,9 +92,47 @@ public async Task<PublishedVersion[]> GetPluginVersionsForDownload(string identi
+            var result = await _httpClient.GetStringAsync(url);
             return JsonConvert.DeserializeObject<PublishedVersion[]>(result, serializerSettings)
                    ?? throw new InvalidOperationException();
         }
+
+        public async Task<PublishedVersion[]> GetInstalledPluginsUpdates(
+            string btcpayVersion,
+            bool includePreRelease,
+            IEnumerable<InstalledPluginRequest> plugins)
+        {
+            var queryString = $"?includePreRelease={includePreRelease}";
+            if (!string.IsNullOrWhiteSpace(btcpayVersion))
+                queryString += $"&btcpayVersion={Uri.EscapeDataString(btcpayVersion)}";
+
+            var json = JsonConvert.SerializeObject(plugins, serializerSettings);
+            using var content = new StringContent(json, Encoding.UTF8, "application/json");
+
+            try
+            {
+                using var resp = await _httpClient.PostAsync($"api/v1/plugins/updates{queryString}", content);
+                resp.EnsureSuccessStatusCode();
+
+                var body = await resp.Content.ReadAsStringAsync();
+                return JsonConvert.DeserializeObject<PublishedVersion[]>(body, serializerSettings)
+                       ?? Array.Empty<PublishedVersion>();
+            }
+            catch (HttpRequestException ex)
+            {
+                _logger.LogWarning(ex, "Failed to check for plugins updates");
+                return Array.Empty<PublishedVersion>();
+            }
+            catch (JsonException ex)
+            {
+                _logger.LogWarning(ex, "Failed to parse plugins updates response");
+                return Array.Empty<PublishedVersion>();
+            }
+            catch (TaskCanceledException ex)
+            {
+                _logger.LogWarning(ex, "Timeout while checking plugins updates");
+                return Array.Empty<PublishedVersion>();
+            }
+        }
```

Do not catch if there is nothing you can do about it. Just let the exception bubble up.